src/charm.py

```diff
@@ -99,86 +102,142 @@ def _configure_pebble(self) -> None:
             self._container.add_layer(self._container_name, layer, combine=True)
             self._container.restart(self._service_name)
 
-    def _configure_upf_information(self):
+    def _configure_upf_information(self) -> None:
         """The `fiveg_n4` relation is not mandatory.
 
-        If it exists it must contain the UPF hostname and port.
+        If it exists, config files are generated and pushed to the workload.
         """
-        if self.model.relations.get(FIVEG_N4_RELATION_NAME):
-            if not self._get_upf_hostname() or not self._get_upf_port():
-                logger.warning("Invalid information in %s integration", FIVEG_N4_RELATION_NAME)
+        if not self.model.relations.get(FIVEG_N4_RELATION_NAME):
+            logger.warning("Relation %s not available", FIVEG_N4_RELATION_NAME)
```

Comment:
Not sure that this should be a warning. When we deploy the control plane and before adding any user planes, it will be the case that there won't be any n4 relation and that's ok. Warnings should be for situations that we are not expecting.